drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,745 @@
+	gc9x01x_transmit(dev, GC9X01X_CMD_GAMMA4, regs->gamma4, sizeof(regs->gamma4));
+	gc9x01x_transmit(dev, GC9X01X_CMD_FRAMERATE, regs->framerate, sizeof(regs->framerate));
+
+	/* Enable Tearing line */
+	gc9x01x_transmit(dev, GC9X01X_CMD_TEON, NULL, 0);
+
+	return 0;
+}
+
+static int gc9x01x_exit_sleep(const struct device *dev)
+{
+	int retval;
+
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_SLPOUT, NULL, 0);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/*
+	 * Exit sleepmode and enable display. 30ms on top of the sleepout time to account for
+	 * any manufacturing defects.
+	 * This is to allow time for the supply voltages and clock circuits stabilize
+	 */
+	k_msleep(GC9X01X_SLEEP_IN_OUT_DURATION_MS + 30U);
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+static int gc9x01x_enter_sleep(const struct device *dev)
+{
+	int retval;
+
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_SLPIN, NULL, 0);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/*
+	 * Exit sleepmode and enable display. 30ms on top of the sleepout time to account for
+	 * any manufacturing defects.
+	 */
+	k_msleep(GC9X01X_SLEEP_IN_OUT_DURATION_MS + 30U);
+
+	return 0;
+}
+#endif
+
+static int gc9x01x_hw_reset(const struct device *dev)
+{
+	const struct gc9x01x_config *config = dev->config;
+
+	if (config->reset.port == NULL) {
+		return -ENODEV;
+	}
+
+	gpio_pin_set_dt(&config->reset, 1U);
+	k_msleep(100);
+	gpio_pin_set_dt(&config->reset, 0U);
+	k_msleep(10);
+
+	return 0;
+}
+
+static int gc9x01x_display_blanking_off(const struct device *dev)
+{
+	LOG_DBG("Turning display blanking off");
+	return gc9x01x_transmit(dev, GC9X01X_CMD_DISPON, NULL, 0);
+}
+
+static int gc9x01x_display_blanking_on(const struct device *dev)
+{
+	LOG_DBG("Turning display blanking on");
+	return gc9x01x_transmit(dev, GC9X01X_CMD_DISPOFF, NULL, 0);
+}
+
+static int gc9x01x_set_pixel_format(const struct device *dev,
+				    const enum display_pixel_format pixel_format)
+{
+	struct gc9x01x_data *data = dev->data;
+
+	int retval;
+	uint8_t tx_data;
+	uint8_t bytes_per_pixel;
+
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		bytes_per_pixel = 2U;
+		tx_data = GC9X01X_PIXFMT_VAL_MCU_16_BIT | GC9X01X_PIXFMT_VAL_RGB_16_BIT;
+	} else if (pixel_format == PIXEL_FORMAT_RGB_888) {
+		bytes_per_pixel = 3U;
+		tx_data = GC9X01X_PIXFMT_VAL_MCU_18_BIT | GC9X01X_PIXFMT_VAL_RGB_18_BIT;
+	} else {
+		LOG_ERR("Unsupported pixel format");
+		return -ENOTSUP;
+	}
+
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_PIXFMT, &tx_data, 1U);
+	if (retval < 0) {
+		return retval;
+	}
+
+	data->pixel_format = pixel_format;
+	data->bytes_per_pixel = bytes_per_pixel;
+
+	return 0;
+}
+
+static int gc9x01x_set_orientation(const struct device *dev,
+				   const enum display_orientation orientation)
+{
+	struct gc9x01x_data *data = dev->data;
+
+	int retval;
+	uint8_t tx_data = GC9X01X_MADCTL_VAL_BGR;
+
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+		/* works 0° */
+		tx_data |= 0;
```

Comment:
maybe drop it and add a comment that that's the default setting?

Reply:
[3af98ab](https://github.com/zephyrproject-rtos/zephyr/commit/3af98ab51e4a42df60a3209ef92480cd397189a3)